### ISSUE TITLE
Register Domain Name

### ISSUE THREAD

**Issue by awesolek2** [HeardLibrary/twine-for-wikicite#71]
Andy is working on this.

**Comment by awesolek2** [comment]
@CliffordAnderson @baskaufs @marjans74 @Talinum @cylew 
Any thoughts on a domain name? I was thinking http://learnwikidata.net (.com or .org are also available)

**Comment by baskaufs** [comment]
In https://github.com/HeardLibrary/twine-for-wikicite/issues/68#issue-916673299 Cliff suggested wikicite4librarians. Personally, I think our videos are broadly applicable enough that something more general like learnwikidata.net might be better.

**Comment by awesolek2** [comment]
Apologies, I didn't see that card on the board. @baskaufs the Twine story is hosted solely on GitHub, right? Barring objections, I'll register learnwikidata.net this afternoon, and configure the DNS to point to the GitHub servers.

**Comment by baskaufs** [comment]
Yes, if you point it to https://heardlibrary.github.io/twine-for-wikicite/, then whatever starting page we set as index.html in the folder will get loaded. That can be either the start of the Twine story, or another non-Twine page if we go with a separate page for the splash page.

**Comment by CliffordAnderson** [comment]
Hey @awesolek2, please do register the domain name you have in mind. Sounds better than the one I proposed!

**Comment by awesolek2** [comment]
Done! I will take 30 minutes or so to process, but I'll let you and @baskaufs know when it has and I've configured the DNS. The .net address is going to cost $11/year, but I see no problem absorbing that through the DiSC AWS account.

**Comment by awesolek2** [comment]
@baskaufs OK, I think I have it set up. When I type learnwikidata.net it is redirecting to github, but I get a 404 for the page. I'm hoping that once you configure the specific page it will work.

**Comment by baskaufs** [comment]
I have been reading up on managing custom domains [here](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site). When I went to the settings page to configure it on the GitHub side, I got this:
![image](https://user-images.githubusercontent.com/5765781/121751624-29e9d580-cad4-11eb-86a3-e4600412e066.png)

I read the [troubleshooting page](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/troubleshooting-custom-domains-and-github-pages) and it says that the CNAME file must contain only the domain name, i.e. `heardlibrary.github.io`. Since that subdomain is used by the whole library, that implies to me that our site will be the only one that will be able to direct to the library's github site.

There must be a way to set up a redirect to an index.html file in a subpath of the subdomain (i.e. `heardlibrary.github.io/twine-for-wikicite`). But I'm out of my element. @CliffordAnderson I think you have more experience with setting up websites like this. Any advice?

**Comment by CliffordAnderson** [comment]
Let's switch on "Enforce HTTPS" for the site, @baskaufs.

**Comment by baskaufs** [comment]
@CliffordAnderson Did you do something to fix this or did it fix itself over time? The custom domain mapping is working now!

No problem, I was planning to do that once the certificate was ready. Actually, I just looked and the box is already checked. 

However, there is still a warning that "www.learnwikidata.net is improperly configured Domain's DNS record could not be retrieved. For more information, see Learn more (InvalidDNSError). We recommend you change this to a CNAME record pointing to heardlibrary.github.io."

**Comment by CliffordAnderson** [comment]
@baskaufs, sorry to bother you. I just did it myself. On your earlier question, @baskaufs, it seems like Github already sorts out the pages just fine. The main github page for [Heard Library](http://heardlibrary.github.io/) is working fine alongside our new [Learn Wikidata](https://learnwikidata.net/) site. But the DNS still shows a configuration error, @awesolek2. 
<img width="908" alt="Screen Shot 2021-06-14 at 9 14 35 AM" src="https://user-images.githubusercontent.com/1919466/121907317-8bd35680-ccf1-11eb-8e55-044c8878754f.png"> Can you maybe check that out?

**Comment by awesolek2** [comment]
OK, I had www.learnwikidata.net set to redirect to heardlibrary.github.io/twine-for-wikicite. Just edited the record to redirect to heardlibrary.github.io. So hopefully that works.

**Comment by baskaufs** [comment]
I'm thinking that the issue is with www.learnwikidat.net vs. learnwikidata.net. learnwikidata.net works, www.learnwikidata.net doesn't. The [troubleshooting page](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/troubleshooting-custom-domains-and-github-pages) says that if properly configured the www subdomain is automatically redirected to the apex domain (learnwikidata.net). Supposedly, you can configure either in the DNS record and one will automatically redirect to the other.

**Comment by CliffordAnderson** [comment]
@awesolek2 I think that'll redirect our site to the main library site. I think you'll need to change it back, Andy. As Steve indicates, we just need to add a subomain for 'www' to the DNS record, I think.

**Comment by baskaufs** [comment]
The other possible issue is that the CNAME file in the docs folder of the repo has "learnwikidata.net" vs. "learnwikidata.net/twine-for-wikicite". I'm wondering what will happen now that Andy made that change. I don't know how the DNS will know to send people to our repo.

**Comment by CliffordAnderson** [comment]
@baskaufs, that's my worry too. But let's see what happens. Maybe there is a routing tool on Github to handle these requests to projects.

**Comment by awesolek2** [comment]
OK I changed it back. I'll look into adding a subdomain.

**Comment by awesolek2** [comment]
OK, I think we are good to go now.

**Comment by baskaufs** [comment]
Here's what I observe at the moment:
1. going to learnwikidata.net is fine.
2. going to www.learnwikidata.net works, but results in this warning:
![image](https://user-images.githubusercontent.com/5765781/121917038-4f582880-ccfa-11eb-92ce-03e544563089.png)
It may be that this certificate problem will solve itself with time since it seems like there is a delay in issuing the certificates.
3. Checking the GitHub pages settings page has a different error message now that I don't really understand:
![image](https://user-images.githubusercontent.com/5765781/121917247-8fb7a680-ccfa-11eb-90d6-85b035c5309f.png)

**Comment by awesolek2** [comment]
Hi Steve. I got that security warning the first time I went to www... too. I accepted the risk and the redirect worked, so I too was assuming it would take a bit for the certificate to authenticate. I just tried again on a different browser, and sure enough, no security threat! So, I think we are good.

**Comment by CliffordAnderson** [comment]
I'm still receiving warnings when trying https://www.learnwikidata.net/ on both Chrome and Firefox. So I think there may still be some tuning at AWS. I'd be happy to consult when you have a chance to hop online. Let's see if we can figure this out together.

**Comment by baskaufs** [comment]
I'm available this afternoon (Mon June 14). I just need to know when so that I can make sure I'm by my computer then.
